From 3ac16688b68eaf71cdabd7d0f97dedc8b94eebe6 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Thu, 4 Jul 2024 09:28:03 +0200
Subject: [PATCH 1200/1695] plat-stm32mp1: fix low power mode selection with
 STPMIC1L

Fix low power mode selection with STPMIC1L and
enable PWR IRQ in STPMIC2 driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4134940755ac53e2b251d048e11b062cb3877f3f plat-stm32mp1: fix low power mode selection with STPMIC1L]
Change-Id: Ibfea63bf674cc5af203ee7517929443a2862db82
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/400303
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk                    | 1 +
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic2.c | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index b499efde9..3cec86417 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -329,6 +329,7 @@ CFG_STM32_I2C ?= y
 CFG_STM32_IWDG ?= y
 CFG_STM32_LPTIMER ?= y
 CFG_STM32_PKA ?= y
+CFG_STM32_PWR_IRQ ?= y
 CFG_STM32_RNG ?= y
 CFG_STM32_RSTCTRL ?= y
 CFG_STM32_RTC ?= y
diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic2.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic2.c
index 1ab2f90f8..48c7d38ab 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic2.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic2.c
@@ -28,9 +28,10 @@ static TEE_Result pmic_regu_pm(enum pm_op op, uint32_t pm_hint,
 			       const struct pm_callback_handle *pm_handle)
 {
 	struct regulator *regulator = pm_handle->handle;
+	unsigned int pwrlvl = PM_HINT_PLATFORM_STATE(pm_hint);
 
 	if (op == PM_OP_SUSPEND)
-		return stm32_pmic2_apply_pm_state(regulator, pm_hint);
+		return stm32_pmic2_apply_pm_state(regulator, pwrlvl);
 
 	return TEE_SUCCESS;
 }
-- 
2.47.2

